operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/managed/DefaultManagedDependentResourceContext.java

```diff
@@ -37,25 +36,13 @@ public <T> T getMandatory(Object key, Class<T> expectedType) {
             + ") is missing or not of the expected type"));
   }
 
-  public DefaultManagedDependentResourceContext setWorkflowExecutionResult(
-      WorkflowReconcileResult workflowReconcileResult) {
-    this.workflowReconcileResult = workflowReconcileResult;
-    return this;
-  }
-
-  public DefaultManagedDependentResourceContext setWorkflowCleanupResult(
-      WorkflowCleanupResult workflowCleanupResult) {
-    this.workflowCleanupResult = workflowCleanupResult;
-    return this;
-  }
-
   @Override
   public Optional<WorkflowReconcileResult> getWorkflowReconcileResult() {
-    return Optional.ofNullable(workflowReconcileResult);
+    return get(RECONCILE_RESULT_KEY, WorkflowReconcileResult.class);
```

Comment:
I agree that it's not super nice but the current way is not great either (casting to the implementation, which is not a given, especially during tests since it breaks with mocks). I actually had a different implementation initially but I would like to remove the reliance on a specific implementation and a cast to it because I don't think it's a nice design.